File: check_balance.js
const fs = require('fs');
const content = fs.readFileSync('d:\\major project\\BharatYatra\\bharat-yatra\\data\\states.js', 'utf8');

let openBraces = 0;
let openBrackets = 0;
let line = 1;

for (let i = 0; i < content.length; i++) {
  const char = content[i];
  if (char === '\n') line++;
  if (char === '{') openBraces++;
  if (char === '}') openBraces--;
  if (char === '[') openBrackets++;
  if (char === ']') openBrackets--;

  if (openBraces < 0) {
    console.log(`Error: Unexpected '}' at line ${line}`);
    process.exit(1);
  }
  if (openBrackets < 0) {
    console.log(`Error: Unexpected ']' at line ${line}`);
    process.exit(1);
  }
}

console.log(`Final balance: Braces: ${openBraces}, Brackets: ${openBrackets}`);

File: check_balance.py
import sys

try:
    with open(r'd:\major project\BharatYatra\bharat-yatra\data\states.js', 'r', encoding='utf-8') as f:
        content = f.read()
except Exception as e:
    print(f"Error reading file: {e}")
    sys.exit(1)

open_braces = 0
open_brackets = 0
line = 1

for i, char in enumerate(content):
    if char == '\n':
        line += 1
    if char == '{':
        open_braces += 1
    elif char == '}':
        open_braces -= 1
    elif char == '[':
        open_brackets += 1
    elif char == ']':
        open_brackets -= 1

    if open_braces < 0:
        print(f"Error: Unexpected '}}' at line {line}")
        sys.exit(1)
    if open_brackets < 0:
        print(f"Error: Unexpected ']' at line {line}")
        sys.exit(1)

print(f"Final balance: Braces: {open_braces}, Brackets: {open_brackets}")
